mountpoint-s3/src/prefetch/feed.rs

```diff
@@ -1,128 +1,266 @@
 use std::{fmt::Debug, ops::Range, sync::Arc};
 
-use async_trait::async_trait;
 use bytes::Bytes;
-use futures::{pin_mut, StreamExt};
-use mountpoint_s3_client::{
-    error::{GetObjectError, ObjectClientError},
-    types::ETag,
-    ObjectClient,
-};
+use futures::task::SpawnExt;
+use futures::{pin_mut, task::Spawn, StreamExt};
+use mountpoint_s3_client::{types::ETag, ObjectClient};
 use mountpoint_s3_crt::checksums::crc32c;
-use tracing::{error, trace};
+use tracing::{debug_span, error, trace, Instrument};
 
 use crate::checksums::ChecksummedBytes;
-use crate::prefetch::{part::Part, part_queue::PartQueueProducer};
+use crate::prefetch::part::Part;
+use crate::prefetch::part_queue::unbounded_part_queue;
+use crate::prefetch::task::RequestTask;
+use crate::store::PrefetchReadError;
 
 /// A generic interface to retrieve data from objects in a S3-like store.
-#[async_trait]
-pub trait ObjectPartFeed<Client: ObjectClient> {
-    /// Get the content of an object in fixed size parts. The parts are pushed to the provided `part_sink`
-    /// and are guaranteed to be contiguous and in the correct order. Callers need to specify a preferred
+pub trait ObjectPartFeed {
```

Comment:
bikeshedding on naming: `ObjectPartStream`?

Reply:
now that we moved the part queue channels to be internal, i think stream makes sense.